crates/neon/src/types_impl/extract/error.rs

```diff
@@ -11,7 +11,42 @@ use crate::{
 
 type BoxError = Box<dyn error::Error + Send + Sync + 'static>;
 
-/// Error returned when a JavaScript value is not the type expected
+/// Error returned when a JavaScript value is not an instance of the expected class.
```

Comment:
This isn't accurate. If it's not the expected class, it's a panic. This is only the error if it's not an object at all.